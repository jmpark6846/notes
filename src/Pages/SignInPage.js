import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import withSizes from "react-sizes";
import { compose } from "recompose";

import { Pane, TextInput, Heading, Button, Text } from "evergreen-ui";

import firebase from "../db";
import { MOBILE_WIDTH } from "../common";

class SignInPage extends Component {
  state = {
    email: "",
    password: "",
    errorMessage: ""
  };
  _handleSignin = async () => {
    try {
      await firebase
        .auth()
        .signInWithEmailAndPassword(this.state.email, this.state.password);
      this.props.history.push("/");
    } catch (error) {
      console.log(error);
      this.setState({ errorMessage: error.message });
    }
  };

  _handleChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  };

  render() {
    return (
      <Pane background="tint1" height="100%" paddingTop={50}>
        <Pane
          background="white"
          marginX="auto"
          paddingY={30}
          paddingX={this.props.isMobile ? 15 : 50}
          elevation={3}
          width={this.props.isMobile ? "calc(100% - 30px)" : 400}
          border="default"
        >
          <Heading size={600} marginBottom={20} textAlign="center">
            로그인
          </Heading>
          <TextInput
            name="email"
            onChange={this._handleChange}
            placeholder="이메일"
            marginBottom={10}
            width="100%"
            value={this.state.email}
            required
          />
          <TextInput
            name="password"
            onChange={this._handleChange}
            placeholder="비밀번호"
            width="100%"
            type="password"
            value={this.state.password}
            required
          />
          <Pane intent="danger" padding={10}>
            <Text color="red">{this.state.errorMessage}</Text>
          </Pane>
          <Pane display="flex" justifyContent="flex-end" marginTop={20}>
            <Button
              marginRight={10}
              onClick={() => this.props.history.push("/signup")}
            >
              회원가입
            </Button>
            <Button
              appearance="primary"
              intent="none"
              onClick={this._handleSignin}
            >
              로그인
            </Button>
          </Pane>
        </Pane>
      </Pane>
    );
  }
}

export default compose(
  withSizes(({ width }) => ({ isMobile: width < MOBILE_WIDTH })),
  withRouter
)(SignInPage);
