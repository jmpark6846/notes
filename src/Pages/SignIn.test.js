import React from "react";
import { render, cleanup, fireEvent } from '@testing-library/react'
import { BrowserRouter as Router} from 'react-router-dom'
import '@testing-library/jest-dom/extend-expect'
import SignInPage from "./SignInPage";

afterEach(cleanup)

test('matches snapshot ', () => {
  const { asFragment } = render(<Router><SignInPage /></Router>)
  expect(asFragment()).toMatchSnapshot()
})


test('should fire post request when click signin button', () => {
  const { queryByPlaceholderText } = render(<Router><SignInPage /></Router>)
  const emailInputElement = queryByPlaceholderText("이메일")
  const passwordInputElement = queryByPlaceholderText("비밀번호")

  emailInputElement.value = "test@test.test"
  fireEvent.change(emailInputElement)

  passwordInputElement.value = "password"
  fireEvent.change(passwordInputElement)

  fireEvent.click(("로그인"))
})
