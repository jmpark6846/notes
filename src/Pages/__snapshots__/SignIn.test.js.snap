// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot  1`] = `
<DocumentFragment>
  <div
    class="📦bg_1vd85pq 📦h_100prcnt 📦pt_50px 📦box-szg_border-box css-nil"
  >
    <div
      class="📦b-top_1px-solid-E4E7EB 📦b-rgt_1px-solid-E4E7EB 📦b-btm_1px-solid-E4E7EB 📦b-lft_1px-solid-E4E7EB 📦bs_17sc08g 📦bg_11ev722 📦ml_auto 📦mr_auto 📦pb_30px 📦pt_30px 📦pl_50px 📦pr_50px 📦w_400px 📦box-szg_border-box css-nil"
    >
      <h2
        class="📦mt_0px 📦mb_20px 📦fnt-sze_20px 📦f-wght_500 📦ln-ht_24px 📦ltr-spc_-0-07px 📦fnt-fam_wm9v6k 📦color_234361 📦txt-algn_center 📦box-szg_border-box"
      >
        로그인
      </h2>
      <input
        aria-invalid="false"
        class="css-5ljhhe 📦color_425A70 📦fnt-fam_b77syt 📦fnt-sze_12px 📦f-wght_400 📦ln-ht_16px 📦ltr-spc_0 📦w_100prcnt 📦h_32px 📦pl_10px 📦pr_10px 📦bblr_3px 📦bbrr_3px 📦btlr_3px 📦btrr_3px 📦mb_10px 📦box-szg_border-box"
        name="email"
        placeholder="이메일"
        required=""
        spellcheck="true"
        type="text"
        value=""
      />
      <input
        aria-invalid="false"
        class="css-5ljhhe 📦color_425A70 📦fnt-fam_b77syt 📦fnt-sze_12px 📦f-wght_400 📦ln-ht_16px 📦ltr-spc_0 📦w_100prcnt 📦h_32px 📦pl_10px 📦pr_10px 📦bblr_3px 📦bbrr_3px 📦btlr_3px 📦btrr_3px 📦box-szg_border-box"
        name="password"
        placeholder="비밀번호"
        required=""
        spellcheck="true"
        type="password"
        value=""
      />
      <div
        class="📦pb_10px 📦pl_10px 📦pr_10px 📦pt_10px 📦box-szg_border-box css-nil"
        intent="danger"
      >
        <span
          class="📦color_red 📦fnt-fam_b77syt 📦fnt-sze_14px 📦f-wght_400 📦ln-ht_20px 📦ltr-spc_-0-05px 📦box-szg_border-box"
        />
      </div>
      <div
        class="📦dspl_flex 📦just-cnt_flex-end 📦mt_20px 📦box-szg_border-box css-nil"
      >
        <button
          class="css-69cngj 📦fnt-fam_b77syt 📦mt_0px 📦fnt-sze_12px 📦f-wght_500 📦ln-ht_32px 📦ltr-spc_0 📦btrr_3px 📦bbrr_3px 📦btlr_3px 📦bblr_3px 📦pt_0px 📦pb_0px 📦pr_16px 📦pl_16px 📦ml_0px 📦mr_10px 📦mb_0px 📦h_32px 📦pst_relative 📦dspl_inline-flex 📦algn-itms_center 📦flx-wrap_nowrap 📦box-szg_border-box"
        >
          회원가입
        </button>
        <button
          class="css-nwvixr 📦fnt-fam_b77syt 📦mt_0px 📦fnt-sze_12px 📦f-wght_500 📦ln-ht_32px 📦ltr-spc_0 📦btrr_3px 📦bbrr_3px 📦btlr_3px 📦bblr_3px 📦pt_0px 📦pb_0px 📦pr_16px 📦pl_16px 📦ml_0px 📦mr_0px 📦mb_0px 📦h_32px 📦pst_relative 📦dspl_inline-flex 📦algn-itms_center 📦flx-wrap_nowrap 📦box-szg_border-box"
        >
          로그인
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
